packages/manager/apps/dedicated/client/app/ip/ip/agoraOrder/ip-ip-agoraOrder.controller.js

```diff
@@ -148,65 +150,104 @@ angular.module('Module.ip.controllers').controller(
+            };
+          });
+        });
+    }
+
     loadIpOffers() {
       this.loading.ipOffers = true;
 
       this.model.params = {};
+      let ipOffersPromise;
 
-      const ipOffersPromise = this.IpAgoraOrder.getIpOffers(
-        this.user.ovhSubsidiary,
-      ).then((ipOffers) => {
-        let ipOfferDetails = ipOffers.map(this.createOfferDto.bind(this));
-        if (this.model.selectedService.type === PRODUCT_TYPES.vps.typeName) {
-          ipOfferDetails = ipOfferDetails
-            .filter(({ productShortName }) =>
-              productShortName.includes('failover'),
-            )
-            .map((offer) => ({
-              ...offer,
-              quantities: range(1, VPS_MAX_QUANTITY + 1),
-            }));
-        }
+      this.isPrivateCloudOffer =
+        get(this.model, 'selectedService.type') ===
+        PRODUCT_TYPES.privateCloud.typeName;
 
-        const ipCountryAvailablePromise = this.IpAgoraOrder.getIpCountryAvailablePromise(
-          this.model.selectedService.serviceName,
-          this.model.selectedService.type,
+      if (this.isPrivateCloudOffer) {
+        ipOffersPromise = this.loadPrivateCloudIpOffers(
+          get(this.model, 'selectedService.serviceName'),
         );
+      } else {
+        ipOffersPromise = this.IpAgoraOrder.getIpOffers(
+          this.user.ovhSubsidiary,
+        ).then((ipOffers) => {
+          let ipOfferDetails = ipOffers.map(this.createOfferDto.bind(this));
+          if (this.model.selectedService.type === PRODUCT_TYPES.vps.typeName) {
+            ipOfferDetails = ipOfferDetails
+              .filter(({ productShortName }) =>
+                productShortName.includes('failover'),
+              )
+              .map((offer) => ({
+                ...offer,
+                quantities: range(1, VPS_MAX_QUANTITY + 1),
+              }));
+          }
 
-        return ipCountryAvailablePromise
-          .then((countries) => {
-            if (countries.length > 0) {
-              const ipOffersByRegion = AgoraIpOrderCtrl.getRegionsOffers(
-                countries,
-              );
-              this.ipOffers = ipOfferDetails
-                .filter(({ productRegion }) =>
-                  ipOffersByRegion.includes(productRegion),
-                )
-                .map((ipOffer) => {
-                  set(
-                    ipOffer,
-                    'countries',
-                    ipOffer.countries.filter(
-                      ({ code }) => countries.indexOf(code.toLowerCase()) > -1,
-                    ),
-                  );
-                  return ipOffer;
-                });
-            } else {
+          const ipCountryAvailablePromise = this.IpAgoraOrder.getIpCountryAvailablePromise(
+            this.model.selectedService.serviceName,
+            this.model.selectedService.type,
+          );
+
+          return ipCountryAvailablePromise
+            .then((countries) => {
+              if (countries.length > 0) {
```

Comment:
```suggestion
              if (isArray(countries) && !isEmpty(countries)) {
```

Reply:
just added a check to ensure countries is defined